routes/ci.js

```diff
@@ -0,0 +1,132 @@
+var request = require("request")
+var repositoryModel = require("../model/repository.js");
+var crypter = require("../util/crypter.js");
+var generator = require("../backend/generator.js");
+var deploy = require("../backend/deploy.js");
+router.get('/register', function (req, res, next) {
+  request({
+    url: `https://api.github.com/users/${req.session.username}/repos?per_page=100`,
+    headers: {
+      'User-Agent': 'request'
+    }
+  },
+  function (error, response, body) {
+    var repos = JSON.parse(body);
+    res.render("index", {
+      repos: repos,
+      ciModal: true
+    });
+  })
+})
+
+router.post('/register', function (req, res, next) {
+
+  var repositoryName = req.body.repository;
+  request({
+    url: `https://api.github.com/repos/${req.session.username}/${repositoryName}/hooks?access_token=${req.session.token}`,
+    headers: {
+      'User-Agent': 'request'
+    }
+  }, function (error, response, body) {
+    var hooks = JSON.parse(body);
+    var hookurl = process.env.HOSTNAME + '/ci/webhook'
+    var isRegistered = false;
+    hooks.forEach(function (hook) {
+      if (hook.config.url == hookurl) {
+        isRegistered = true
+      }
+    })
+    if (!isRegistered) {
+      request({
+        url: `https://api.github.com/repos/${req.session.username}/${repositoryName}/hooks?access_token=${req.session.token}`,
+        headers: {
+          'User-Agent': 'request'
+        },
+        method: 'POST',
+        json: {
+          name: "web",
+          active: true,
+          events: [
+            "push"
+          ],
+          config: {
+            url: hookurl,
+            content_type: "json"
+          }
+        }
+      }, function(error, response, body) {
+        repositoryModel.newRepository(repositoryName,
+          req.session.username,
+          req.session.githubId,
+          crypter.encrypt(req.session.token),
+          req.session.email)
+          .then(function(result) {
+            res.render("index", {
+              showMessage: true,
+              messages: `Thanks for registering with Yaydoc.Here after Documentation will be pushed on each commit`
+            })
+          })
+          .catch(function(err) {
+            next({
+              status: 500,
+              message: 'something went wrong with db'
+            })
+          })
+      })
+    } else {
+      res.render("index", {
+        showMessage: true,
+        messages: `Yaydoc was already integrate with this repository`
+      })
+    }
+  })
+})
+
+router.post('/webhook', function(req, res, next) {
+  var event = req.get('X-GitHub-Event')
+  switch (event) {
+    case "push":
+      repositoryModel.findOneRepository(
+        {
+          githubId: req.body.repository.owner.id,
+          name: req.body.repository.name
+        }
+      ).
+      then(function(result) {
+        var data = {
+          email: result.email,
+          gitUrl: req.body.repository.clone_url,
+          docTheme: "alabaster",
```

Comment:
Don't hard code `alabaster`. Instead set it to be an empty string. `generate.sh` will handle the rest